https://github.com/ceph/ceph-csi/actions/runs/17289568981/job/49168477608?pr=5547
@Acedus go-test is still failing
```
=== CONT  TestFetchIP/client.4305_2001:0db8:85a3:0000:0000:8a2e:0370:7334:0/422650892
    fencing_test.go:95: expected error false but got failed to extract IP address, incorrect format: address client.4305 2001:0db8:85a3:0000:0000:8a2e:0370:7334:0/422650892: too many colons in address
    fencing_test.go:99: expected IP 2001:db8:85a3::8a2e:370:7334 but got 
--- FAIL: TestFetchIP (0.00s)
    --- FAIL: TestFetchIP/client.4305_172.21.9.34:0/422650892 (0.00s)
    --- PASS: TestFetchIP/#00 (0.00s)
    --- FAIL: TestFetchIP/client.24152_v1:100.64.0.7:0/3658550259 (0.00s)
    --- FAIL: TestFetchIP/client.4305_2001:0db8:85a3:0000:0000:8a2e:0370:7334:0/422650892 (0.00s)
FAIL
```

--- a/internal/csi-addons/networkfence/fencing_test.go
+++ b/internal/csi-addons/networkfence/fencing_test.go
@@ -68,7 +68,7 @@ func TestFetchIP(t *testing.T) {
 			expectedErr: false,
 		},
 		{
-			clientInfo:  "client.4305 2001:0db8:85a3:0000:0000:8a2e:0370:7334:0/422650892",
+			clientInfo:  "client.4305 [2001:0db8:85a3:0000:0000:8a2e:0370:7334]:0/422650892",
 			expectedIP:  "2001:db8:85a3::8a2e:370:7334",
 			expectedErr: false,
 		},

--- a/internal/util/util.go
+++ b/internal/util/util.go
@@ -302,24 +302,18 @@ func GetVolumeContext(parameters map[string]string) map[string]string {
 // Handles both IPv4 and IPv6 addresses.
 // Example: "10.244.0.1:0/2686266785" returns "10.244.0.1".
 func ParseClientIP(addr string) (string, error) {
-	// Attempt to extract the IP address using a regular expression
-	// the regular expression aims to match either a complete IPv6
-	// address or a complete IPv4 address follows by any prefix (v1 or v2)
-	// if exists
-	// (?:v[0-9]+:): this allows for an optional prefix starting with "v"
-	// followed by one or more digits and a colon.
-	// The ? outside the group makes the entire prefix section optional.
-	// (?:[0-9a-fA-F]{1,4}:){7}[0-9a-fA-F]{1,4}: this allows to check for
-	// standard IPv6 address.
-	// |: Alternation operator to allow matching either the IPv6 pattern
-	// with a prefix or the IPv4 pattern.
-	// '(?:\d+\.){3}\d+: This part matches a standard IPv4 address.
-	re := regexp.MustCompile(`(?:v[0-9]+:)?([0-9a-fA-F]{1,4}(:[0-9a-fA-F]{1,4}){7}|(?:\d+\.){3}\d+)`)
-	ipMatches := re.FindStringSubmatch(addr)
+	versionPrefixRe := regexp.MustCompile(`(?:v\d+:)`)
+	for hostPair := range strings.SplitSeq(addr, " ") {
+		hostPair = versionPrefixRe.ReplaceAllString(hostPair, "")
 
-	if len(ipMatches) > 0 {
-		ip := net.ParseIP(ipMatches[1])
-		if ip != nil {
+		host, _, err := net.SplitHostPort(hostPair)
+		if err != nil {
+			// We might fail if there is no port specified, in that case continue
+			// as if it's just an IP address and try to parse it.
+			host = hostPair
+		}
+
+		if ip := net.ParseIP(host); ip != nil {
 			return ip.String(), nil
 		}
 	}

--- a/internal/util/util_test.go
+++ b/internal/util/util_test.go
@@ -300,8 +300,14 @@ func TestParseClientIP(t *testing.T) {
 		},
 		{
 			name:    "IPv6 address",
-			addr:    "2001:0db8:85a3:0000:0000:8a2e:0370:7334:0/2686266785",
+			addr:    "[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:0/2686266785",
 			want:    "2001:db8:85a3::8a2e:370:7334",
+			wantErr: false,
+		},
+		{
+			name:    "Compressed IPv6 address",
+			addr:    "[fd98::4]:0/2686266785",
+			want:    "fd98::4",
 			wantErr: false,
 		},
 		{

